03week/ticTacToe.js

```diff
@@ -14,34 +15,85 @@ let board = [
+// let rowZero = board[0]
+// let rowOne = board[1]
+// let rowTwo = board[2]
+
 function printBoard() {
   console.log('   0  1  2');
   console.log('0 ' + board[0].join(' | '));
   console.log('  ---------');
   console.log('1 ' + board[1].join(' | '));
   console.log('  ---------');
   console.log('2 ' + board[2].join(' | '));
-}
+};
 
 function horizontalWin() {
-  // Your code here
-}
+  //checks for the 3 cases of a horizontal win for each player
+  if ((board[0][0] == 'X' && board[0][1] == 'X' && board[0][2] == 'X') ||
+    (board[1][0] == 'X' && board[1][1] == 'X' && board[1][2] == 'X') ||
+    (board[2][0] == 'X' && board[2][1] == 'X' && board[2][2] == 'X') ||
+    (board[0][0] == 'O' && board[0][1] == 'O' && board[0][2] == 'O') ||
+    (board[1][0] == 'O' && board[1][1] == 'O' && board[1][2] == 'O') ||
+    (board[2][0] == 'O' && board[2][1] == 'O' && board[2][2] == 'O')) {
+    console.log(`${playerTurn} wins!`)
+    return true
+  } else {//if no win found then it exits to verticalWin func
+    return verticalWin();
+  }
+
+};
 
 function verticalWin() {
-  // Your code here
-}
+  //checks for the 3 cases of a vert win for each player
+  if ((board[0][0] === 'X' && board[1][0] === 'X' && board[2][0] === 'X') ||
+    (board[0][1] === 'X' && board[1][1] === 'X' && board[2][1] === 'X') ||
+    (board[0][2] === 'X' && board[1][2] === 'X' && board[2][2] === 'X') ||
+    (board[0][0] === 'O' && board[1][0] === 'O' && board[2][0] === 'O') ||
+    (board[0][1] === 'O' && board[1][1] === 'O' && board[2][1] === 'O') ||
+    (board[0][2] === 'O' && board[1][2] === 'O' && board[2][2] === 'O') ) {
+    console.log(`${playerTurn} wins!`)
+    return true
+  } else {//if not win found then it exits to diagonlWin func
+    return diagonalWin();
+  }
+};
 
 function diagonalWin() {
-  // Your code here
-}
-
-function checkForWin() {
-  // Your code here
-}
+  //checks for last 2 cases of win, diagonally, for each player
+  if ((board[0][0] === 'X' && board[1][1] === 'X' && board[2][2] === 'X') ||
+    (board[0][2] === 'X' && board[1][1] === 'X' && board[2][0] === 'X') ||
+    (board[0][0] === 'O' && board[1][1] === 'O' && board[2][2] === 'O') ||
+    (board[0][2] === 'O' && board[1][1] === 'O' && board[2][0] === 'O')) {
+    console.log(`${playerTurn} wins!`)
+    return true
+  }
+};
+
+
+function checkForWin(board) {
+  //8 winning cases are checked with 3 seperate functions, starting with horizontal
+  return horizontalWin(board);
+};
+
+function switchPlayerTurn() {
+//checks for which playerTurn it is and alternates back and forth each turn
+  if (playerTurn === 'X'){
+    playerTurn = 'O'
+  } else {
+    playerTurn = 'X'
+  }
+};
 
 function ticTacToe(row, column) {
-  // Your code here
-}
+  //reassigning value of board array and index with the playerTurn of 'X' or 'O'
+  board[row][column] = playerTurn
+
+  checkForWin(board);
+  switchPlayerTurn();//switches player turn ONLY after board is set and win is checked for
```

Comment:
this switches the player turn no matter what, you should wrap checkForWin in an if statement